src/server/string_family.cc

```diff
@@ -565,7 +567,15 @@ void StringFamily::Set(CmdArgList args, ConnectionContext* cntx) {
   const auto result{SetGeneric(cntx, sparams, key, value)};
 
   if (result == OpStatus::OK) {
-    return builder->SendStored();
+    if (!(sparams.flags & SetCmd::SET_GET)) {
```

Comment:
1. please reverse the condition and the blocks.
2. my question still holds (maybe I missed the answer): when SET_GET is set and prev_val is  null?